db/migrate/20240718105718_create_integration_error_details.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class CreateIntegrationErrorDetails < ActiveRecord::Migration[7.1]
+  def change
+    create_table :integration_error_details, id: :uuid do |t|
+      t.references :owner, type: :uuid, polymorphic: true, null: false, index: true
+      t.references :error_producer, type: :uuid, polymorphic: true, index: true, null: false
```

Comment:
I was deciding between STI, delegated_type and polymorphism
Initially i ended up with polymorphic `:integration` as it would have highlighted that payment_provider and integration are the same instance, and later, when we'd do the refactoring we'd get rid of this relation at all (having simple integration_id).
(also because integration error should have the same behaviour for different types of integrations and the same fields)
But then I thought about making this model even more generic, if we're talking not only about payment_providers and integrations, but if suddenly we have something else, and in this case something, where the error happened is something that created that error, so then this is the error producer.
(at the same time I'm still thinking that technically all external integrations should follow pattern of integrations, and would be perfect to end up with only integration_id...)

Reply:
(but then should it be something like `third_party_error_details`?

Reply:
Hey Anna thanks a lot for thinking this through and caring.

I'd go as far as calling the table `error_details` and let anyone add their error there. 
It makes sense that most errors would come from 3rd parties (because our errors are supposed to be handled 😄) so I'd add a polymorphic relation to integration **but make it optional.**

```rb
create_table :error_details, id: :uuid do |t|
  t.references :owner, type: :uuid, polymorphic: true, null: false, index: true
  t.references :integration, type: :uuid, polymorphic: true, index: true, null: true
  # ...
end
```

I also strongly believe we should migration the `payment_providers` table content to `integrations` and keep using the same PaymentProvider classes/code. But let's talk about this separately. 👍